### ISSUE TITLE
Site Editor: Saving buttons in the left panel trigger a full focus loss

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#55550]
### Description

When saving changes in the Site editor, by using the 'Save' button in the left panel, a full focus loss occurs.

Actually, the Save button is entirely replaced by another element, the 'Saving' button. As such, when using a keyboard, the currently focused element is removed from the DOM thus triggering the focus loss.

See this pattern:

https://github.com/WordPress/gutenberg/blob/ade7e648250795e0ea4b89a6a3b905b5c61bce97/packages/edit-site/src/components/save-hub/index.js#L147-L168

Conditionally rendering two different components when the component is an interactive, focused, element is less than ideal.

This pattern should either be prohibited in the entire codebase or it should be abstracted in a component to manage focus.

![blue](https://github.com/WordPress/gutenberg/assets/1682452/96e5cdd6-7baf-4394-81ad-3d281159f4d4)

![gray low contrast](https://github.com/WordPress/gutenberg/assets/1682452/11d9963f-f52a-42a4-9d72-839ec8e818dc)

Additionally:
- I'm not sure I understand why some views in the editor use the 'gray' saving button and other views use the 'blue' (primary) one.
- The gray button has an `aria-expanded="false"` attribtute that looks unnecessary.
- The grey button gets an `aria-disabled="true"` attribute, while tbe blue button gets an `aria-disabled="false"`.
- The grey button has a pointless aria label, as it just repeats the visible text: `<button ... aria-label="Saving">Saving</button>`
- The blue button gets a HTML `disabled` attribute in addition to`aria-disabled="true"`.
-  This is not only redundant but it also makes the button not focusable, thus contributing to the focus loss. Worth reminding once again that the editor uses a well established pattern since years which is using`aria-disabled="true"`and noop-ing the control instead of using `disabled`.

### Step-by-step reproduction instructions

- Go to the site editor > Styles 
- Change theme style.
- Use the keyboard and navigate to the blue Save button.
- Press Enter.
- Observe the button gets entirely replaced and focus is lost.

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by afercia** [comment]
Animated GIF to illustrate the full focus loss. Tested with Safari + VoiceOver.

After saving, pressing the Tab key again restarts the tab sequence from the document root:

![focus loss](https://github.com/WordPress/gutenberg/assets/1682452/9cd3cce9-69b8-45c2-a3b3-04dc04797d53)

**Comment by afercia** [comment]
> Conditionally rendering two different components when the component is an interactive, focused, element is less than ideal.

Theis has been refactored in https://github.com/WordPress/gutenberg/pull/60077 and doesn't conditionally render two different components any longer. As such, there is no focus loss any longer.

Currently, the button oopens the EntitiesSaved modal dialog. When the modal dialog closes, focus is correctly returned to the save button.

Closing as fixed elsewhere. The other points reported in this issue are minor and some of them outdated.
